src/main/java/lotto/LottoGame.java

```diff
@@ -1,26 +1,43 @@
 package lotto;
 
 import java.util.Collection;
+import java.util.function.Supplier;
 
 import lotto.model.LottoMatcher;
 import lotto.model.LottoNumberGenerator;
 import lotto.model.LottoNumbers;
 import lotto.model.MatchResult;
+import lotto.model.Number;
 import lotto.model.Payment;
 import lotto.view.InputView;
 import lotto.view.OutputView;
 
 public class LottoGame {
+    private static final String ERROR_MESSAGE_PREFIX = "[ERROR] ";
+
     private LottoGame() {
     }
 
     public static void start() {
-        Payment payment = InputView.readPaymentHandlingException();
+        Payment payment = handleException(InputView::readPayment);
         Collection<LottoNumbers> lottoNumbersCollection =
```

Comment:
```suggestion
        Collection<LottoNumbers> lottoNumbers =
```
변수명에 컬렉션을 써줘야 하는 이유가 있나요?

Reply:
lottoNumbers를 여러 개 모아놓은 것이라 저렇게 표현했습니다. lottoNumberss 처럼 ss를 2개 붙힐 수도 없고 그렇다고 lottoNumbers라고만 쓰면 LottoNumbers 객체 한 개만 있는 것과 구분이 되지 않을 것 같습니다.

Reply:
lottoNumbers 라는 클래스이름을 바꿔볼 수 있지 않을까요?
의미 있는 클래스명 없을까요? 로또번호 묶음을 뭐라 부를 수 있을까요?

Reply:
로또가 번호 6개를 갖고 있는거니까 차라리 LottoNumbers를 Lotto라고 변경하고 위에값을 lottos라고 바꾸면 해결될 것 같다는 생각이 드네요^^;